lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -0,0 +1,239 @@
+        post[:amount] = amount(money).to_f
+        # Visanet Peru expects a 9-digit numeric purchaseNumber
+        post[:purchaseNumber] = options[:order_id]
+        post[:externalTransactionId] = options[:order_id]
+        post[:currencyId] = CURRENCY_CODES[options[:currency] || currency(money)]
+      end
+
+      def add_payment_method(post, payment_method)
+        post[:firstName] = payment_method.first_name
+        post[:lastName] = payment_method.last_name
+        post[:cardNumber] = payment_method.number
+        post[:cvv2Code] = Integer(payment_method.verification_value, 10)
+        post[:expirationYear] = format(payment_method.year, :four_digits)
+        post[:expirationMonth] = format(payment_method.month, :two_digits)
+      end
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email]
+        antifraud = {}
+        billing_address = options[:billing_address] || options[:address]
+        if (billing_address)
+          antifraud[:billTo_street1] = billing_address[:address1]
+          antifraud[:billTo_city] = billing_address[:city]
+          antifraud[:billTo_state] = billing_address[:state]
+          antifraud[:billTo_country] = billing_address[:country]
+          antifraud[:billTo_postalCode]    = billing_address[:zip]
+        end
+        post[:antifraud] = antifraud
+      end
+
+      def add_device_fingerprint_data(post, options)
+        post[:antifraud][:deviceFingerprintId] = options[:device_fingerprint_id]
+      end
+
+      def add_merchant_define_data(post, options)
+        if (merchantDefineData = options[:merchant_define_data])
+          post[:antifraud][:merchantDefineData] = merchantDefineData
+        end
+      end
+
+      def commit(action, params, options)
+        case action
+        when "authorize"
+          url = base_url() + "/" + options[:merchant_id]
+          method = :post
+        when "capture"
+          url = base_url() + "/" + options[:merchant_id] + "/deposit/" + options[:purchaseNumber]
+          method = :put
+        when "void"
+          url = base_url() + "/" + options[:merchant_id] + "/void/" + options[:purchaseNumber]
+          method = :put
+        when "cancel"
+          url = base_url() + "/" + options[:merchant_id] + "/cancelDeposit/" + options[:purchaseNumber]
+          method = :put
+        end
+        begin
+          raw_response = ssl_request(method, url, post_data(action, params), headers)
+          response = parse(raw_response)
+        rescue ResponseError => e
+          raw_response = e.response.body
+          response_error(raw_response)
+        rescue JSON::ParserError
+          unparsable_response(raw_response)
+        else
+          Response.new(
+            success_from(response),
+            message_from(response),
+            response,
+            :test => test?,
+            :authorization => action + "|" + (response["merchantId"] || '') + "|" + (response["externalTransactionId"] || ''),
```

Comment:
Can we move this concatenation logic to its own method, just to keep it clearly delineated?